src/computability/epsilon_NFA.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2021 Fox Thomson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fox Thomson
+-/
+
+import computability.NFA
+
+/-!
+# Epsilon Nondeterministic Finite Automata
+This file contains the definition of an epsilon Nondeterministic Finite Automaton (ε_NFA), a state
+machine which determines whether a string (implemented as a list over an arbitrary alphabet) is in a
+regular set by evaluating the string over every possible path, also having access to ε-transitons,
+which can be followed without reading a character.
+Note that this definition allows for Automaton with infinite states, a `fintype` instance must be
+supplied for true ε_NFA's.
```

Comment:
```suggestion
Since this definition allows for Automata with infinite states, a `fintype` instance must be
supplied for true `ε_NFA`'s.
```

This fact is probably worth mentioning in the doc string for `ε_NFA` as well.